litmus-portal/frontend/src/views/Analytics/ApplicationDashboards/Table/index.tsx

```diff
@@ -183,33 +210,81 @@ const DashboardTable: React.FC = () => {
                 ? sortNumAsc(y, x)
                 : sortNumDesc(y, x);
             }
-            if (filter.sortData.dashboardType.sort) {
-              const x = a.db_type;
-              const y = b.db_type;
-              return filter.sortData.dashboardType.ascending
-                ? sortAlphaAsc(x, y)
-                : sortAlphaDesc(x, y);
-            }
-            if (filter.sortData.dataSourceType.sort) {
-              const x = a.ds_type;
-              const y = b.ds_type;
-              return filter.sortData.dataSourceType.ascending
-                ? sortAlphaAsc(x, y)
-                : sortAlphaDesc(x, y);
-            }
-            if (filter.sortData.agent.sort) {
-              const x = a.cluster_name;
-              const y = b.cluster_name;
-              return filter.sortData.agent.ascending
-                ? sortAlphaAsc(x, y)
-                : sortAlphaDesc(x, y);
-            }
             return 0;
           })
     : [];
 
+  useEffect(() => {
+    if (dataSourceList && dataSourceList.ListDataSource) {
+      const activeDataSources: ListDataSourceResponse[] =
+        dataSourceList.ListDataSource.filter(
+          (dataSource) => dataSource.health_status === 'Active'
+        ) ?? [];
+      if (activeDataSources.length) {
+        setActiveDataSourceAvailable(true);
+      }
+    }
+  }, [dataSourceList]);
+
   return (
     <div className={classes.root}>
+      {!activeDataSourceAvailable && !loadingDataSources && (
+        <blockquote className={classes.warningBlock}>
+          <Typography className={classes.warningText} align="left">
+            {dataSourceList?.ListDataSource.length
+              ? t(
+                  'analyticsDashboard.applicationDashboardTable.warning.noActiveDataSource'
+                )
+              : t(
+                  'analyticsDashboard.applicationDashboardTable.warning.noAvailableDataSource'
+                )}
+          </Typography>
+          <div className={classes.warningActions}>
+            {dataSourceList?.ListDataSource.length ? (
+              <>
+                <TextButton
+                  onClick={() => tabs.changeAnalyticsDashboardTabs(3)}
+                  variant="highlight"
+                  className={classes.warningButton}
+                >
+                  <Typography
+                    className={classes.buttonText}
+                    style={{ fontWeight: 500 }}
+                  >
+                    {t(
+                      'analyticsDashboard.applicationDashboardTable.warning.configureExisting'
+                    )}
+                  </Typography>
+                </TextButton>
+                <Typography className={classes.orText}>
+                  {t('analyticsDashboard.applicationDashboardTable.warning.or')}
+                </Typography>
+              </>
+            ) : (
+              <></>
```

Comment:
same

Reply:
same

Reply:
This one is not necessary use &&

Reply:
0 getting rendered for negative case will recheck though

Reply:
0 is a falsely value && will catch that